amethyst_renderer/src/tex_animation.rs

```diff
@@ -0,0 +1,178 @@
+    /// this should play initially.
+    ///
+    /// ## Returns
+    /// If the `starting_animation` wasn't found then this will return `None`.  This could be due
+    /// to one of a few things:
+    ///
+    /// - The `sprite_sheet_data` hadn't loaded yet.  Handles aren't valid in their first frame of
+    /// existence.
+    /// - An animation with the name from `starting_animation` wasn't found in `sprite_sheet_data`.
+    /// - The `sprite_sheet_data` handle wasn't made with the `sheet_storage` AssetStorage.
+    pub fn new(playback_speed: f32, sprite_sheet_data: SpriteSheetDataHandle, sheet_storage: &AssetStorage<SpriteSheetData>, starting_animation: &str) -> Option<SpriteSheetAnimation> {
+        sheet_storage.get(&sprite_sheet_data).and_then(|data| data.animation_mapping.get(starting_animation).map(|i| *i)).map(|current_animation| {
+            SpriteSheetAnimation {
+                playback_speed,
+                sprite_sheet_data,
+                current_animation,
+                current_frame: 0,
+                frame_timer: Duration::from_secs(0),
+            }
+        })
+
+    }
+
+    /// Sets the animation to the named case-sensitive animation from the `SpriteSheetData`.
+    ///
+    /// ## Returns
+    ///
+    /// If the name provided can't be found in the `SpriteSheetData` for this component then this
+    /// will return `Err` otherwise it will return `Ok`.
+    pub fn set_animation(&mut self, sheet_storage: &AssetStorage<SpriteSheetData>, animation: &str) -> Result<(), ()> {
+        let animation = sheet_storage.get(&self.sprite_sheet_data).and_then(|data| data.animation_mapping.get(animation).map(|i| *i));
+        match animation {
+            Some(animation) => {
+                self.current_animation = animation;
+                self.current_frame = 0;
+                self.frame_timer = Duration::from_secs(0);
+                Ok(())
+            }
+            None => {
+                Err(())
+            }
+        }
+    }
+
+    /// Retrieve frame information for the current frame.
+    ///
+    /// ## Returns
+    ///
+    /// Most of the time this should return `Some`.  If it doesn't then either the
+    /// `SpriteSheetDataHandle` used in initialization or the `sheet_storage` in this function are
+    /// invalid.  Because handles take at least one frame to become valid after being loaded the
+    /// handle is the most likely point of failure.
+    pub fn current_frame(&self, sheet_storage: &AssetStorage<SpriteSheetData>) -> Option<Frame> {
+        sheet_storage.get(&self.sprite_sheet_data).map(|data| data.frames[data.animations[self.current_animation][self.current_frame]].clone())
+    }
+
+    /// Advance the current animation by the time given.  Normally this will be called by the
+    /// engine by default using the regularly running game clock.  This function is made public so
+    /// that you can manually advance the animation if you require.
```

Comment:
How user can disable automatic animation?
The system is added by bundle.
What if user wants only some entities to be manually animated?

Reply:
Not built into the current implementation, but I'll give it some thought.